File: truevfs-ext/truevfs-ext-insight/src/main/java/net/java/truevfs/ext/insight/I5tIoStatisticsMXBean.java
/*
 * Copyright (C) 2005-2012 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight;

import javax.annotation.concurrent.ThreadSafe;
import net.java.truevfs.ext.insight.stats.IoStatistics;

/**
 * An MXBean interface for {@linkplain IoStatistics I/O statistics}.
 *
 * @author Christian Schlichtherle
 */
@ThreadSafe
public interface I5tIoStatisticsMXBean {
    int    getReadBytesPerOperation();
    long   getReadBytesTotal();
    long   getReadKilobytesPerSecond();
    long   getReadNanosecondsPerOperation();
    long   getReadNanosecondsTotal();
    long   getReadOperations();
    int    getReadThreadsTotal();
    String getSubject();
    String getTimeCreatedDate();
    long   getTimeCreatedMillis();
    String getTimeUpdatedDate();
    long   getTimeUpdatedMillis();
    int    getWriteBytesPerOperation();
    long   getWriteBytesTotal();
    long   getWriteKilobytesPerSecond();
    long   getWriteNanosecondsPerOperation();
    long   getWriteNanosecondsTotal();
    long   getWriteOperations();
    int    getWriteThreadsTotal();

    void rotate();
}

File: truevfs-ext/truevfs-ext-insight/src/main/scala/net/java/truevfs/ext/insight/I5tIoStatisticsMXBean.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight

import javax.annotation.concurrent._

/**
 * An MXBean interface for [[net.java.truevfs.ext.insight.stats.IoStatistics]].
 *
 * @author Christian Schlichtherle
 */
@ThreadSafe
trait I5tIoStatisticsMXBean {

  def getReadBytesPerOperation: Int
  def getReadBytesTotal: Long
  def getReadKilobytesPerSecond: Long
  def getReadNanosecondsPerOperation: Long
  def getReadNanosecondsTotal: Long
  def getReadOperations: Long
  def getReadThreadsTotal: Int
  def getSubject: String
  def getTimeCreatedDate: String
  def getTimeCreatedMillis: Long
  def getTimeUpdatedDate: String
  def getTimeUpdatedMillis: Long
  def getWriteBytesPerOperation: Int
  def getWriteBytesTotal: Long
  def getWriteKilobytesPerSecond: Long
  def getWriteNanosecondsPerOperation: Long
  def getWriteNanosecondsTotal: Long
  def getWriteOperations: Long
  def getWriteThreadsTotal: Int

  def rotate()
}
